filebeat/input/netmetrics/udp.go

```diff
@@ -0,0 +1,239 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package netmetrics
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/rcrowley/go-metrics"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+	"github.com/elastic/elastic-agent-libs/monitoring/adapter"
+)
+
+// UDPMetrics captures udp related metrics.
+type UDPMetrics struct {
+	done chan struct{}
+
+	lastPacket time.Time
+
+	device         *monitoring.String // name of the device being monitored
+	packets        *monitoring.Uint   // number of packets processed
+	bytes          *monitoring.Uint   // number of bytes processed
+	bufferLen      *monitoring.Uint   // configured read buffer length
+	rxQueue        *monitoring.Uint   // value of the rx_queue field from /proc/net/udp{,6} (only on linux systems)
+	drops          *monitoring.Uint   // number of udp drops noted in /proc/net/udp{,6}
+	arrivalPeriod  metrics.Sample     // histogram of the elapsed time between packet arrivals
+	processingTime metrics.Sample     // histogram of the elapsed time between packet receipt and publication
+}
+
+func NewUDPMetrics(reg *monitoring.Registry, device string, buflen uint64, poll time.Duration, log *logp.Logger) *UDPMetrics {
```

Comment:
Why has the option for a nil collector gone away?

Reply:
answered in TCP comment